k2/python/csrc/torch/v2/ragged_arc.cu

```diff
@@ -58,6 +64,32 @@ RaggedArc::RaggedArc(
   // TODO: we also need to pass the name of extra_labels and ragged_labels.
 }
 
+RaggedArc::RaggedArc(const RaggedArc &src, const Ragged<Arc> &arcs,
+                     torch::Tensor arc_map) {
+  fsa = arcs;
+  for (auto iter = src.tensor_attrs.begin(); iter != src.tensor_attrs.end();
+      ++iter) {
+    auto value = IndexSelectFunction::apply(iter->second, arc_map, 0);
+    SetAttr(iter->first, value);
+  }
+
+  for (auto iter = src.ragged_tensor_attrs.begin();
+      iter != src.ragged_tensor_attrs.end(); ++iter) {
+    // Only integer types ragged attributes are supported now
+    K2_CHECK_EQ(iter->second.any.GetDtype(), kInt32Dtype);
+    auto value = const_cast<RaggedAny &>(iter->second).Index(arc_map, 0, false);
+    SetAttr(iter->first, value.first);
+  }
+
+  for (auto iter = src.other_attrs.begin(); iter != src.other_attrs.end();
+      ++iter) {
+    SetAttr(iter->first, iter->second);
+  }
+  // populate scores
+  Scores();
+  IndexSelectScoresFunction::apply(*this, src.Scores(), arc_map);
```

Comment:
This line can alse be `scores = IndexSelectFunction::apply(src.Scores(), arc_map, 0)` (the same with line 72).  The purpose of using another function `IndexSelectScoresFunction` is to share the same memory between `fsa.scores` and `scores`, is it right?